src/Connect.js

```diff
@@ -1,68 +1,441 @@
+   * @param    {Object}      [opts]                       optional parameters
+   * @param    {Object}      [opts.network='rinkeby']     network config object or string name, ie. { id: '0x1', registry: '0xab5c8051b9a1df1aab0149f8b0630848b7ecabf6', rpcUrl: 'https://mainnet.infura.io' } or 'kovan', 'mainnet', 'ropsten', 'rinkeby'.
+   * @param    {Object}      [opts.provider=HttpProvider] Provider used as a base provider to be wrapped with uPort connect functionality
+   * @param    {String}      [opts.accountType]           Ethereum account type: "general", "segregated", "keypair", "devicekey" or "none"
+   * @param    {Boolean}     [opts.isMobile]              Configured by default by detecting client, but can optionally pass boolean to indicate whether this is instantiated on a mobile client
+   * @param    {Boolean}     [opts.storage=true]          When true, object state will be written to local storage on each state cz-conventional-change
+   * @param    {Function}    [opts.transport]             Configured by default by detecting client, but can optionally pass boolean to indicate whether this is instantiated on a mobile client
+   * @param    {Function}    [opts.mobileTransport]       Configured by default by detecting client, but can optionally pass boolean to indicate whether this is instantiated on a mobile client
+   * @param    {Object}      [opts.muportConfig]          Configuration object for muport did resolver. See [muport-did-resolver](https://github.com/uport-project/muport-did-resolver)
+   * @param    {Object}      [opts.ethrConfig]            Configuration object for ethr did resolver. See [ethr-did-resolver](https://github.com/uport-project/ethr-did-resolver)
+   * @param    {Object}      [opts.registry]              Configuration for uPort DID Resolver (DEPRACATED) See [uport-did-resolver](https://github.com/uport-project/uport-did-resolver)
+   * @return   {Connect}                                  self
    */
-
   constructor (appName, opts = {}) {
-    super(appName, opts)
-    this.uriHandler = opts.uriHandler || openQr
-    this.mobileUriHandler = opts.mobileUriHandler || mobileUriHandler
-    this.closeUriHandler = opts.closeUriHandler || (this.uriHandler === openQr ? closeQr : undefined)
+    // Config
+    this.appName = appName || 'uport-connect-app'
+    this.network = network.config.network(opts.network)
+    this.provider = opts.provider || new HttpProvider(this.network.rpcUrl)
+    this.accountType = opts.accountType
+    this.isOnMobile = opts.isMobile === undefined ? isMobile() : opts.isMobile
+    this.storage = opts.storage === undefined ? true : opts.storage
+    // Transports
+    this.transport = opts.transport || connectTransport(appName)
+    this.mobileTransport = opts.mobileTransport || transport.url.send()
+    this.onloadResponse = transport.url.getResponse()
+    this.PubSub = PubSub
+    transport.url.listenResponse((err, payload) => {
```

Comment:
@zachferland why are you using `transport.url.listenResponse` here? How can someone implement the same functionality using other transport?